### ISSUE TITLE
Use SemVer?

### ISSUE THREAD

**Issue by HerHde** [forrestguice/SuntimesWidget#21]
Ahoi Forrest,
now I've got a proposal: How about using a [semantic versioning scheme](http://semver.org/)?

For example, the version bump from 0.1.0 to 0.1.1 could have been a 0.2.0, because you've added i18n.

Greetings
~HerHde

**Comment by forrestguice** [comment]
I'm on the fence. I read up on it and I really like the idea of semantic versioning - it seems great for libraries and public interfaces - I need to version the app's data source interface. ...but I'm not sure how to apply it to the application itself.

I have a lot of trouble figuring out just what kind of change constitutes bumping the minor number vs the patch - in some cases a change sounds big, gets its own line in the release notes (i.e. "adds support for i18n"), but its really a handful of minor fixes (which was the case w/ the recent i18n stuff). In retrospect I should have probably bumped it from 0.1.0 to 0.2.0, but at the time I was thinking something like "this doesn't really change any of the functionality and most people won't notice anything different", so I bumped the patch instead. More recently I've added the option to override the locale - the changes to the code were more significant (and I'm hoping haven't broken anything) so I'm going to bump the minor number this time. On the other hand, does pulling a single hardcoded value into strings constitute a change in the app's public interface? I don't think it should, but I could argue that its technically a "backwards compatible api change" and not a fix.

So basically I have no idea how to version this thing, and the more I think about it, the less certain I am. LOL

I got a kick out of reading http://sentimentalversioning.org .. it would seem I'm currently using "romantic versioning" ;)

**Comment by HerHde** [comment]
I would interpret it IMHO as a
MAJOR, when features are removed OR when the user is forced to revisit the app OR reconfigure some stuff (e.g. the settings are altered).
MINOR, when you add some features, like i18n, IPv6 support, a theme editor, a little bit more than just one option which only adds to existing components (I hope it is understandable).
PATCH, when you add little options, which keep the former defaults OR just add some details to the GPS settings for example OR when a translation is updated.

The romantic versioning sounds nice, too. Either way I would bump MAJOR versions, if the user SHOULD take a look at the app. Also I probably would then show them a changelog or a simpler information on the changes.

P.S.: I'm sorry for not answering atm, I'm having a lot on my plate, at least until next week.

**Comment by forrestguice** [comment]
...reflecting on this lately.

I think I'm happy so far w/ the distinction the app makes between minor and patch versions .. that adding any kind of feature/functionality bumps the minor version, while fixes or minor UI changes bump the patch. That leaves the role of the major version - at what point will the app no longer be major version `v0` and become `v1`.

To me bumping the major version seems arbitrary (like a form of branding), but I think #122 (drop support for api 10) helps make a distinction. Any changes to the minsdk constitutes a major change (effectively dropping support for an entire category of devices). When the app eventually drops support for api10 it will need to bump the major version.

Until such a 'breaking change' occurs I'm going to continue bumping the minor version .. v0.7.0, v0.8.0, v0.9.0, v0.10.0, v0.11.0 etc, as features are added (additional widgets, golden hour, moon functionality, etc).

**Comment by HerHde** [comment]
Yes, this is indeed a good approach, especially considering the API support and breaking changes in general. It also seems to be a valid interpretation of the SemVer in case of an app.

Do it ;-)
